### PR TITLE
Decimal numbers

### PR DESCRIPTION
**Pull request by toikarin-vincit**
# Hitas Pull Request

# Description

c63e5b3: backend: start using decimal number for prices
 - old system used integers in many places, including calculations.
   let's start using decimal numbers to get more precise results :)
   this also simplifies things a bit as we don't need to constantly
   check if a given value happens to be an integer or a decimal
   number because we wanted to be consistent with the old system
 - this could mean that maximum price calculation reuslts change
   a bit as rounding is taken into considiration only after the
   final calculated maximum price, not during the calculations

---

4e72f64: backend: add a single missing invalid value test
 - invalid apartment_share_of_housing_company_loans_date

---

79730d2: backend: small import fix

---

32ff939: backend: IndexFactory: set high value so generated values are not always 0

---

0f9202f: backend: max price test json refactor
 - use max price calculation to generate the json value
   instead of creating the JSON with hard-coded structure.
   This way we don't need to constantly update the structure
   in the tests when structure changes.

---

## Pull request checklist

Check the boxes for each DoD items that has been completed:

- **Frontend**
    - [x] Changes have been tested
- **Backend**
    - [x] Changes have been tested
    - [x] Automatic tests has been added
    - [x] OpenAPI definitions have been updated
    - [x] Database migrations will work in test environment
    - [ ] Oracle migration has been updated

## Test plan

- Frontend is not updated. OpenAPI definitions have been updated to return `number` instead of `integer` and JSON responses d the same :)

## Tickets

This pull request resolves all or part of the following ticket(s): HT-306

### CODE REVIEW

**Review by ranta — COMMENTED** [review]
Was the idea to change every price related integer to a decimal? In that case there are still some missing. Otherwise looks good to me :)

**Comment by toikarin-vincit** [comment]
> Was the idea to change every price related integer to a decimal? In that case there are still some missing. Otherwise looks good to me :)



Which ones? I checked and could not find any. Only questionable is `value_per_square_meter` but it is always an integer

**Comment by ranta** [comment]
> Which ones? I checked and could not find any. Only questionable is `value_per_square_meter` but it is always an integer

Yes, i was referring to that, but also `PricesSerializer` and `ConstructionPrices` have multiple price fields that are still integers

**Comment by toikarin-vincit** [comment]
> > Which ones? I checked and could not find any. Only questionable is `value_per_square_meter` but it is always an integer
> 
> Yes, i was referring to that, but also `PricesSerializer` and `ConstructionPrices` have multiple price fields that are still integers

Ah. I was thinking it was a bit strange I didn't need to update that many tests :grin: Fixed

**Review by ranta — APPROVED** [review]
LGTM 🚀

File: backend/hitas/calculations/max_price.py [hunk 1]
```diff
@@ -342,11 +342,11 @@ def calculate_index(
         apartment_surface_area=apartment.surface_area,
     )
 
+    print(type(basic_price))
```
[the remark this answers is not in the document]
Comment (toikarin-vincit) [reply backend/hitas/calculations/max_price.py]: oof, added bd74945

**Pull request merged**